openmrs/src/main/java/org/motechproject/openmrs/domain/Observation.java

```diff
@@ -211,19 +278,23 @@ public JsonElement serialize(ObservationValue src, Type typeOfSrc, JsonSerializa
     }
 
     /**
-     * Implementation of the {@link JsonDeserializer} interface for the
-     * {@link Observation.ObservationValue} class.
-     */
+    * Implementation of the {@link JsonDeserializer} interface for the
+    * {@link Observation.ObservationValue} class.
+    */
     public static class ObservationValueDeserializer implements JsonDeserializer<ObservationValue> {
-
         @Override
-        public ObservationValue deserialize(JsonElement json, Type typeOfT, JsonDeserializationContext context) throws JsonParseException {
-            String value = json.getAsString();
-            JsonObject valueObject = new JsonObject();
-            valueObject.addProperty(DISPLAY_KEY, value);
+        public ObservationValue deserialize (JsonElement json, Type typeOfT, JsonDeserializationContext context) throws JsonParseException {
+            ObservationValue observationValue = new ObservationValue(null);
+
+            if (json.isJsonObject()) {
+                JsonObject jsonObject = json.getAsJsonObject();
+                observationValue.setDisplay(jsonObject.get(DISPLAY_KEY).getAsString());
+            } else if (json.isJsonPrimitive()) {
+                JsonPrimitive jsonPrimitive = json.getAsJsonPrimitive();
+                observationValue.setDisplay(jsonPrimitive.getAsString());
+            }
 
-            return (ObservationValue) JsonUtils.readJson(valueObject.toString(), ObservationValue.class);
+            return observationValue;
```

Comment:
I see that you are using this deserializer only in method not connected with this data provider. What's the reason ?